kv/memberlist/memberlist_client.go

```diff
@@ -201,6 +203,12 @@ func (cfg *KVConfig) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix string) {
 	f.BoolVar(&cfg.ClusterLabelVerificationDisabled, prefix+"memberlist.cluster-label-verification-disabled", mlDefaults.SkipInboundLabelCheck, "When true, memberlist doesn't verify that inbound packets and gossip streams have the cluster label matching the configured one. This verification should be disabled while rolling out the change to the configured cluster label in a live memberlist cluster.")
 	f.DurationVar(&cfg.BroadcastTimeoutForLocalUpdatesOnShutdown, prefix+"memberlist.broadcast-timeout-for-local-updates-on-shutdown", 10*time.Second, "Timeout for broadcasting all remaining locally-generated updates to other nodes when shutting down. Only used if there are nodes left in the memberlist cluster, and only applies to locally-generated updates, not to broadcast messages that are result of incoming gossip updates. 0 = no timeout, wait until all locally-generated updates are sent.")
 
+	// For our use cases, we don't need a very fast detection of dead nodes. Since we use a TCP transport
+	// and we open a new TCP connection for each packet, we prefer to reduce the probe frequency and increase
+	// the timeout compared to defaults.
+	f.DurationVar(&cfg.ProbeInterval, prefix+"memberlist.probe-interval", 5*time.Second, "Probe a random node every this interval. This setting is also the total timeout for the direct + indirect probes.")
+	f.DurationVar(&cfg.ProbeTimeout, prefix+"memberlist.probe-timeout", 2*time.Second, "Timeout for the direct probe.")
```

Comment:
If we want to change it for tests only, let's not expose this as configurable fields.

Reply:
You mean put the new configs in `KVConfig` but not exposed as flags?

Reply:
🤔. I could just bump the polling time to 7s. That would be an easier fix, if we don't want to expose the config at all

Reply:
Here, this should be better 😄

Reply:
:) Thanks.

Reply:
> You mean put the new configs in KVConfig but not exposed as flags?

Yes, this is what I meant. Also hide from yaml by using `yaml:"-"` tag.